assembly/src/assembler/mod.rs

```diff
@@ -574,12 +574,14 @@ impl Assembler {
         let proc_body_id = if num_locals > 0 {
             // for procedures with locals, we need to update fmp register before and after the
             // procedure body is executed. specifically:
-            // - to allocate procedure locals we need to increment fmp by the number of locals
-            // - to deallocate procedure locals we need to decrement it by the same amount
-            let num_locals = Felt::from(num_locals);
+            // - to allocate procedure locals we need to increment fmp by 4 times the number of
+            //   locals
+            // - to deallocate procedure locals we need to decrement it by the same amount We leave
+            // 4 elements between locals to properly support reading and writing words to locals.
+            let locals_frame = Felt::from(num_locals * WORD_SIZE as u16);
```

Comment:
nit: there is a period missing on line 579 and indentation missing on line 580.

Also, we should probably mention it very clearly in the docs that when we specify number of locals for a procedure, we are actually specifying the number of words. An alternative could be to specify the number of elements, but require them to be in multiples of 4 (this would be an easy compile-time error to catch).

Reply:
One thing we should do is make sure that the first procedure local is always word-aligned. Otherwise, it will be difficult to procedure code to know if they can use locals procedures which expect word-aligned memory pointers.

Reply:
Done.

> One thing we should do is make sure that the first procedure local is always word-aligned.

It is, since `FMP_MIN` is word-aligned, and procedures are only allowed to increment the `fmp` pointer by a multiple of 4